fix: remove noisy chromatic diffs due to dynamic dates and form link

kevin9foong opened the pull request:
## Problem
<!-- What problem are you trying to solve? What issue does this close? -->

There are noisy chromatic test differences due to dynamic dates for save draft and form link for printable response. 

## Solution
<!-- How did you solve the problem? -->
Mock with constant values to prevent diffs between builds. 

**Breaking Changes** 
<!-- Does this PR contain any backward incompatible changes? If so, what are they and should there be special considerations for release? -->
No - this PR is backwards compatible

kevin9foong committed:
chore: use constant form link to prevent noisy diffs for chromatic

kevin9foong committed:
chore: add mock date time for test

kevin9foong commented on frontend/src/features/public-form/PublicFormProvider.tsx:
```diff
@@ -1388,6 +1388,12 @@ export const PublicFormProvider = ({
+  const MOCK_CONSTANT_LAST_SAVED_DATETIME_STRING_FOR_STORYBOOK_TO_HAVE_NO_DIFFS = format(new Date('2025-11-14T12:00:00.000Z'), DATE_TIME_FORMAT_STRING)
+  const actualLastSavedDateTimeString = draftSubmission?.lastUpdated ? format(new Date(draftSubmission.lastUpdated), DATE_TIME_FORMAT_STRING) : undefined
+  // RATIONALE: Replace with constant string if actual last saved date time string exists during storybook visual tests to prevent noisy diffs.
+  const draftLastSavedDateTimeString = actualLastSavedDateTimeString && isTest ? MOCK_CONSTANT_LAST_SAVED_DATETIME_STRING_FOR_STORYBOOK_TO_HAVE_NO_DIFFS : actualLastSavedDateTimeString
```
note that if the actual last saved date time does not exist (ie, no draft saved), we will not override with the mocked constant date time string

kevin9foong committed:
chore: fix lint formatting

KenLSM commented on frontend/src/features/public-form/PublicFormProvider.tsx:
```diff
@@ -1388,6 +1388,18 @@ export const PublicFormProvider = ({
+  const draftLastSavedDateTimeString =
+    actualLastSavedDateTimeString && isTest
+      ? MOCK_CONSTANT_LAST_SAVED_DATETIME_STRING_FOR_STORYBOOK_TO_HAVE_NO_DIFFS
+      : actualLastSavedDateTimeString
```
Would this also make sense? 

```suggestion
  // move these two to top-level constants
  const DATE_TIME_FORMAT_STRING = 'do MMM yyyy, h:mm:ss a'
  const MOCK_CONSTANT_LAST_SAVED_DATETIME = '2025-11-14T12:00:00.000Z'

  const lastSavedDateTime = draftSubmission?.lastUpdated && isTest
    ? MOCK_CONSTANT_LAST_SAVED_DATETIME
    : draftSubmission?.lastUpdated
    
  const lastSavedDateTimeString = lastSavedDateTime
    ? format(new Date(lastSavedDateTime), DATE_TIME_FORMAT_STRING)
    : undefined
```

kevin9foong replied:
Yeah - this is more concise, will apply the fix

KenLSM commented on frontend/src/features/admin-form/responses/IndividualResponsePage/PrintableResponse.tsx:
```diff
@@ -181,6 +181,11 @@ const PrintableResponseRows = ({
 
+const isTest = import.meta.env.STORYBOOK_NODE_ENV === 'test'
+// RATIONALE: Prevent noisy diffs from being detected during storybook snapshot comparisons due to dynamic form link.
+const MOCK_CONSTANT_FORM_LINK_FOR_STORYBOOK_TO_HAVE_NO_DIFFS =
```
```suggestion
const MOCK_CONSTANT_FORM_LINK =
```

kevin9foong replied:
Applied the fix

KenLSM reviewed (approved):
LGTM. 2 nits.

kevin9foong committed:
chore: simplify save draft datetime mocking

Co-authored-by: Ken Lee Shu Ming <ken@open.gov.sg>

Update frontend/src/features/admin-form/responses/IndividualResponsePage/PrintableResponse.tsx

Co-authored-by: Ken Lee Shu Ming <ken@open.gov.sg>

chore: simplify date time mocking
